### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by Nicogene**
This PR reduce the n of warnings from ~70 to 5 on clang.

Please review code.

### CODE REVIEW

File: src/libYARP_dev/src/MapGrid2D.cpp [hunk 1]
```diff
@@ -654,7 +652,6 @@ bool  MapGrid2D::crop (int left, int top, int right, int bottom)
     new_map_occupancy.resize(right-left,bottom-top);
     new_map_flags.resize(right-left,bottom-top);
 
-    size_t original_width = m_map_occupancy.width();
```
Comment (Nicogene) [inline src/libYARP_dev/src/MapGrid2D.cpp]: @randaz81 this variable was unused, is it a leftover or an hidden bug?

File: src/carriers/portmonitor_carrier/lua/lua_swig.h [hunk 2]
```diff
@@ -549,14 +549,14 @@ SWIG_MangledTypeQueryModule(swig_module_info *start,
   swig_module_info *iter = start;
   do {
     if (iter->size) {
-      register size_t l = 0;
-      register size_t r = iter->size - 1;
+      size_t l = 0;
+      size_t r = iter->size - 1;
```
Comment (drdanz) [inline src/carriers/portmonitor_carrier/lua/lua_swig.h]: This is an auto-generated file:

```
/* ----------------------------------------------------------------------------
 * This file was automatically generated by SWIG (http://www.swig.org).
 * Version 2.0.7
 * 
 * This file is not intended to be easily readable and contains a number of 
 * coding conventions designed to improve portability and efficiency. Do not make
 * changes to this file unless you know what you are doing--modify the SWIG 
 * interface file instead. 
 * ----------------------------------------------------------------------------- */
```

Let's investigate how this file was generated and re-generate it with a recent swig version instead!
CC @apaikan

File: src/libYARP_dev/src/IJoypadController.cpp [hunk 3]
```diff
@@ -439,7 +439,7 @@ bool yarp::dev::IJoypadController::parseActions(const yarp::os::Searchable& cfg,
         }
         if(!keyvalue.get(0).isInt32()                 ||
             keyvalue.get(0).asInt32() < 0             ||
-            keyvalue.get(0).asInt32() > buttonCount-1 ||
+           (unsigned int) keyvalue.get(0).asInt32() > buttonCount-1 ||
```
Comment (drdanz) [inline src/libYARP_dev/src/IJoypadController.cpp]: @aerydna YOU DESERVE THIS! :hocho: :skull_and_crossbones: 

 :laughing:

**Comment by drdanz** [comment]
Merged, thanks.

**Pull request merged**